File: C#/3_Insertion Sort/Insertion Sort.cs
using System;

namespace Algorithm {

    class InsertionSort {

        public static void insertionSort(int[] arr){
            int i, j;
            for(i = 1; i < arr.Length; i++){
                j = i;
                while(j > 0 && arr[j-1] > arr[j]){
                    int temp = arr[j];
                    arr[j] = arr[j-1];
                    arr[j-1] = temp;
                    j--;
                }
            }
        }

        public static void printArray(int[] arr){
            for(int i = 0; i < arr.Length; i++){
                Console.Write(arr[i] + " ");
            }
        }

        public static void Main(string[] args) {
            int[] arr = {5, 2, 42, 6, 1, 3, 2};
            insertionSort(arr);
            printArray(arr);
        }
    }
}

File: Ruby/3_Insertion Sort/Insertion Sort.rb
def insertionSort(arr)
	for i in 1...arr.length
		j = i
		while j > 0 && arr[j-1] > arr[j]
			arr[j], arr[j-1] = arr[j-1], arr[j]
			j -= 1
		end
	end
end

nums = [3, 2, 1, 5, 8, 6, 9, 4, 7]
insertionSort(nums)
p nums